### PR TITLE
Update NuGet.config

### PR DESCRIPTION
**Pull request by pranavkm**
@wtgodbe

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -3,6 +3,7 @@
   <packageSources>
     <clear />
     <add key="dotnet-core" value="https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json" />
```
Comment (dougbu) [inline NuGet.config]: @mmitche @wtgodbe where are we on `dotnet-core` versus `dotnet-eng` versus `dotnet-tools`? Do we need two of those three these days?

File: NuGet.config [hunk 2]
```diff
@@ -3,6 +3,7 @@
...
+    <add key="dotnet3.1" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3.1/nuget/v3/index.json" /> 
     <add key="dotnet3.1-transport" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet3.1-transport/nuget/v3/index.json" /> 
     <add key="dotnet-tools" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-tools/nuget/v3/index.json" />
     <add key="aspnet-extensions" value="https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json" />
```
Comment (dougbu) [inline NuGet.config]: This PR should also remove the following
``` xml
    <add key="aspnet-extensions" value="https://dotnetfeed.blob.core.windows.net/aspnet-extensions/index.json" />
    <add key="aspnet-aspnetcore-tooling" value="https://dotnetfeed.blob.core.windows.net/aspnet-aspnetcore-tooling/index.json" />
```

FYI (or a reminder) the following can be removed after the next 3.1 release
``` xml
    <add key="aspnet-stable" value="https://dotnetfeed.blob.core.windows.net/dotnet-core-3-1-rtm-014727/index.json" />
```

**Review by dougbu — CHANGES_REQUESTED** [review]

File: NuGet.config (hunk 1 above)
Comment (wtgodbe) [reply NuGet.config]: We can remove `dotnet-tools` in favor of `dotnet-eng`. There might be some straggling packages still in `dotnet-core` that prevent us from removing it - rel/3.1 still has it:

https://github.com/dotnet/aspnetcore/blob/64a7f37e8afb060f3dbf8a679e1f7112a4558562/NuGet.config#L14

Comment (mmitche) [reply NuGet.config]: dotnet-core can probalby be removed in favor of "dotnet5" and "dotnet5-transport"

Comment (mmitche) [reply NuGet.config]: (fyi I removed it from websdk recently)

File: NuGet.config (hunk 2 above)
Comment (pranavkm) [reply NuGet.config]: We might be fine removing aspnet-stable too, we only reference 3.1.2 packages in this branch. I'll let the CI tell me if that worked

**Comment by dougbu** [comment]
@pranavkm should we resolve the conflicts in favour of your branch and retry the build?

**Comment by pranavkm** [comment]
```

F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error : Unable to find package Microsoft.DotNet.Arcade.Sdk with version (= 1.0.0-beta.20124.2)
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error :   - Found 178 version(s) in dotnet-tools [ Nearest version: 5.0.0-beta.19467.5 ]
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error :   - Found 1 version(s) in blazor-wasm [ Nearest version: 5.0.0-beta.19557.14 ]
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error :   - Found 1 version(s) in dotnet3.1-transport [ Nearest version: 5.0.0-beta.19531.16 ]
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error :   - Found 0 version(s) in nuget.org
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error :   - Found 0 version(s) in dotnet3.1
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error : F:\workspace\_work\1\s\.dotnet\sdk\3.1.102\Sdks\Microsoft.DotNet.Arcade.Sdk\Sdk not found. Check that a recent enough .NET Core SDK is installed and/or increase the version specified in global.json.
F:\workspace\_work\1\s\artifacts\toolset\restore.proj : error MSB4236: The SDK 'Microsoft.DotNet.Arcade.Sdk' specified could not be found.
```

What feed does this come from?

**Comment by wtgodbe** [comment]
@pranavkm you need `dotnet-eng`, and can remove `dotnet-tools`

https://github.com/dotnet/aspnetcore/blob/64a7f37e8afb060f3dbf8a679e1f7112a4558562/NuGet.config#L15

**Comment by pranavkm** [comment]
Good to go

**Review by JunTaoLuo — APPROVED** [review]

**Comment by mmitche** [comment]
Oh man that's an awesome reduction.

**Comment by mmitche** [comment]
LGTM

**Review by wtgodbe — APPROVED** [review]

**Comment by mmitche** [comment]
This might have an impact on the restore perf for this repo too.

**Pull request merged**